rust/azure_iot_operations_services/src/azure_device_registry.rs

```diff
@@ -2,5 +2,1347 @@
+    /// The name of the management group.
+    pub name: String,
+    /// URI or type definition id of the management group
+    pub type_ref: Option<String>,
+}
+
+/// Represents action management group
+#[derive(Clone, Debug)]
+pub struct AssetManagementGroupAction {
+    /// Configuration for the action.
+    pub action_configuration: Option<String>,
+    /// Type of action.
+    pub action_type: AssetManagementGroupActionType,
+    /// The name of the action.
+    pub name: String,
+    /// The target URI for the action.
+    pub target_uri: String,
+    /// The timeout for the action.
+    pub time_out_in_seconds: Option<u32>,
+    /// The MQTT topic for the action.
+    pub topic: Option<String>,
+    /// URI or type definition id of the management group action
+    pub type_ref: Option<String>,
+}
+
+/// Represnts the stream of the asset.
+#[derive(Clone, Debug)]
+pub struct AssetStream {
+    /// Destinations for a stream.
+    pub destinations: Vec<EventsAndStreamsDestination>, // if None, we can represent as empty vec. Can currently only be length of 1
+    /// The name of the stream.
+    pub name: String,
+    /// The configuration for the stream.
+    pub stream_configuration: Option<String>,
+    /// URI or type definition id of the stream
+    pub type_ref: Option<String>,
+}
+
+/// The data point in the event.
+#[derive(Clone, Debug)]
+pub struct AssetEventDataPoint {
+    /// The configuration for the data point in the event.
+    pub data_point_configuration: Option<String>,
+    /// The data source for the data point in the event.
+    pub data_source: String,
+    /// The name of the data point in the event.
+    pub name: String,
+}
+
+// TODO: turn into rust enums for which of these options can correlate to which destination enums
+/// The configuration for the destination
+#[derive(Clone, Debug)]
+pub struct DestinationConfiguration {
+    /// The key of the destination configuration.
+    pub key: Option<String>,
+    /// The description of the destination configuration.
+    pub path: Option<String>,
+    /// The MQTT `QoS` setting for the destination configuration.
+    pub qos: Option<Qos>,
+    /// The MQTT retain setting for the destination configuration.
+    pub retain: Option<Retain>,
+    /// The MQTT topic for the destination configuration.
+    pub topic: Option<String>,
+    /// The MQTT TTL setting for the destination configuration.
+    pub ttl: Option<u64>,
+}
+
+// ~~~~~~~~~~~~~~~~~~~Asset Status DTDL Equivalent Structs~~~~~~~
+#[derive(Clone, Debug, Default)]
+/// Represents the observed status of an asset.
+pub struct AssetStatus {
+    /// The configuration of the asset.
+    pub config: Option<StatusConfig>,
+    /// A collection of datasets associated with the asset.
+    pub datasets_schema: Option<Vec<AssetDatasetEventStreamStatus>>,
+    /// A collection of events associated with the asset.
+    pub events_schema: Option<Vec<AssetDatasetEventStreamStatus>>,
```

Comment:
either have `schema` in all of these variable names or none of them (management groups and streams)

Reply:
probably remove schema since it's more of a schema and/or status